src/txmempool.cpp

```diff
@@ -1245,3 +1536,169 @@ std::vector<CTxMemPool::txiter> CTxMemPool::GatherClusters(const std::vector<uin
     }
     return clustered_txs;
 }
+
+void Cluster::AddTransaction(const CTxMemPoolEntry& entry, bool sort)
+{
+    m_chunks.emplace_back(entry.GetModifiedFee(), entry.GetTxSize());
+    m_chunks.back().txs.emplace_back(entry);
+    entry.m_cluster = this;
+    ++m_tx_count;
+    if (sort) Sort();
+    return;
+}
+
+void Cluster::RemoveTransaction(const CTxMemPoolEntry& entry)
+{
+    m_chunks[entry.m_loc.first].txs.erase(entry.m_loc.second);
+
+    // Chunk (or cluster) may now be empty, but this will get cleaned up
```

Comment:
I wouldn't say it's safe to ignore, but the idea is that we often want to be able to batch deletions, and then clean things up in one pass.  So the call sites should all be dealing with this issue and ensuring that we always clean up at some point.

(This is definitely an area where I expect that we'll be re-engineering all this logic and trying to come up with a better abstraction layer so that this is more robust and easier to think about!)